feature/image-processing/src/main/java/com/rempawl/image/processing/ImageProcessingViewModel.kt

```diff
@@ -4,66 +4,156 @@ import androidx.lifecycle.ViewModel
-            // todo lift logic to StateCase
-            either {
-                val texts = async { textDetectionUseCase.call(inputImage).bind() }
-                val objects = async { objectDetectionUseCase.call(inputImage).bind() }
-                texts.await() to objects.await()
+        fileUtils.getInputImage(imageUri)
+            .onSuccess { inputImage ->
+                // todo lift logic to StateCase
+                processInputImage(inputImage, imageUri)
             }
-                .onSuccess { (texts, objects) ->
-                    _state.update {
-                        it.copy(
-                            isProgressVisible = false,
-                            showError = false,
-                            detectedTextObjects = texts,
-                            detectedObjects = objects,
-                            imageState = ImageState(
-                                height = inputImage.height,
-                                width = inputImage.width,
-                                uri = imageUri
-                            )
+            .onError {
+                _state.update {
+                    it.copy(
+                        showError = true,
+                        isProgressVisible = false
+                    )
+                } // todo AppError class end different error types
+            }
+    }
+
+    private suspend fun processInputImage(
+        inputImage: InputImage,
+        imageUri: String,
+    ) = coroutineScope {
+        either {
+            val texts = async { textDetectionUseCase.call(inputImage).bind() }
+            val objects = async { objectDetectionUseCase.call(inputImage).bind() }
+            texts.await() to objects.await()
+        }
+            .onSuccess { (texts, objects) ->
+                _state.update {
+                    it.copy(
+                        isProgressVisible = false,
+                        showError = false,
+                        detectedTextObjects = texts,
+                        detectedObjects = objects,
+                        imageState = ImageState(
+                            height = inputImage.height,
+                            width = inputImage.width,
+                            uri = imageUri
                         )
-                    }
+                    )
                 }
-                .onError {
-                    _state.update {
-                        it.copy(
-                            isProgressVisible = false,
-                            imageState = ImageState(),
-                            showError = true,
-                            detectedTextObjects = emptyList(),
-                            detectedObjects = emptyList()
-                        )
-                    }
+            }
+            .onError {
+                _state.update {
+                    it.copy(
+                        isProgressVisible = false,
+                        imageState = ImageState(),
```

Comment:
would be good to make ImageState sealed class with state.success and state.error